.circleci/config.yml

```diff
@@ -455,6 +455,7 @@ jobs:
     resource_class: gpu.small
     environment:
       image_name: "pytorch/manylinux-cuda101"
+      PYTHON_VERSION: << parameters.python_version >>
```

Comment:
I was working on the same issue on torchaudio, and although I understand that this solves the issue, I wonder if there is a better way.
The problem here is `environment: image_name: "..."` overwrites the `environment` defined in `binary_common`. It seems to me that the ideal implementation is that `image_name` should be appended to the set of `environment` key-value pairs defined in `binary_common`, so that the downstream jobs do not have to re-define them. But with quick searching on YAML syntax, I cannot find such a way.
On the other hand, `binary_common` was defined for binary build process (probably by @seemethere ) and I just re-used it in unit test jobs without knowing much about it in the initial implementation of CCI job configurations (in torchaudio, which was later ported to torchvision). 
Environment values like `UPLOAD_CHANNEL` and `PYTORCH_VERSION` have become an essential part of unit test scripts in the previous release work, https://github.com/pytorch/vision/pull/2846, so ideally we want environment variables to be defined in one place and we want them to be automagically propagated.
The use of Docker adds extra layer to this as it seems inevitable to manually pass environment values to the container which runs the test.